examples/games/mafia_werewolf/play_mafia.qnt

```diff
@@ -0,0 +1,185 @@
+// -*- mode: Bluespec; -*-
+
+module mafia {
+    const PLAYERS : Set[str]
+    // Types for roles and life states of players
+    type Role = Mafia | Citizen
+    type LifeState = Alive | Dead
+    type Phase = Day | Night
+    type PlayerFeatures = {
+        role: Role,
+        status: LifeState,
+        voted: bool  // Indicates whether the player has voted
+    }
+    var player_to_features: str -> PlayerFeatures
+    var votes_by_player: str -> int
+    var game_phase: Phase
+    type Status = Pending | Done(Role)
+    var game_status: Status
+    /// Check if all alive players have voted
+    val all_voted = PLAYERS.filter(p => player_to_features.get(p).status == Alive)
+                      .forall(p => player_to_features.get(p).voted == true)
+    /// Check if there are any Mafia players left
+    val has_mafia = PLAYERS.exists(p => player_to_features.get(p).role == Mafia)
+    /// Check if there are any Citizen players left
```

Comment:
You say "left" here in the comment but, since this doesn't filter for alive players, it will always be the same (i.e. it won't become false if all Mafia/Citzens die).
```suggestion
    /// Check if there are any Mafia players
    val has_mafia = PLAYERS.exists(p => player_to_features.get(p).role == Mafia)
    /// Check if there are any Citizen players
```